cli/azd/pkg/project/container_helper.go

```diff
@@ -0,0 +1,70 @@
+package project
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/environment"
+	"github.com/benbjohnson/clock"
+)
+
+type ContainerHelper struct {
+	env   *environment.Environment
+	clock clock.Clock
+}
+
+func NewContainerHelper(env *environment.Environment, clock clock.Clock) *ContainerHelper {
+	return &ContainerHelper{
+		env:   env,
+		clock: clock,
+	}
+}
+
+func (ch *ContainerHelper) LoginServer(ctx context.Context) (string, error) {
+	loginServer, has := ch.env.Values[environment.ContainerRegistryEndpointEnvVarName]
+	if !has {
+		return "", fmt.Errorf(
+			"could not determine container registry endpoint, ensure %s is set as an output of your infrastructure",
+			environment.ContainerRegistryEndpointEnvVarName,
+		)
+	}
+
+	return loginServer, nil
+}
+
+func (ch *ContainerHelper) RemoteImageTag(ctx context.Context, serviceConfig *ServiceConfig) (string, error) {
+	loginServer, err := ch.LoginServer(ctx)
+	if err != nil {
+		return "", err
+	}
+
+	localTag, err := ch.LocalImageTag(ctx, serviceConfig)
```

Comment:
Not a fan that we're generating dynamic values inside a function that seems like it just fetches information.

assuming `t` increases by `1` with each statement:

```golang
ch.LocalImageTag() // *azd-deploy-<time_n>
ch.RemoteImageTag() // <loginServer>/*azd-deploy-<time_n+1>
```

Do you think we can think about creating testable `helper` methods if we need to test the logic, and avoid the helper object?

Reply:
This helper was introduced to support multiple container based service targets and share code between them.  If we are concerned about the timestamp generation we can append the remote tag portion to the existing local tag.